templates/common/openstack/units/afterburn-hostname.service.yaml

```diff
@@ -3,7 +3,7 @@ enabled: true
 contents: |
   [Unit]
   Description=Afterburn Hostname
-  Before=kubelet.service
+  Before=network-online.target kubelet.service crio.service
```

Comment:
If afterburn is getting it's info from nova-metadata service, shouldn't it have it's networking stack ready?

Reply:
yeah, for sure. but afterburn can also get these values from the config drive. in addition to this PR I want to restore my patches for the config drive in the installer.

Reply:
We also tear down the network before exiting the initramfs - this allows us to support configuring the network via Ignition.

Reply:
Since multi-user.target wants `network-online.target` and that requires networking to be up [1], changing this to be
```
[Unit]
Before=node-valid-hostname.service
```

And 
```
[Install]
WantedBy=network-online.target
```
Would accomplish the goal and work with the "hardnames are hard work." 

[1] https://github.com/openshift/machine-config-operator/blob/master/templates/common/_base/units/node-valid-hostname.service.yaml